client/src/appsupport.res

```diff
@@ -0,0 +1,310 @@
+  let isDesktop = Belt.Option.isSome(Js.String.match_(%re("/DarkLang\/Editor/"), navigatorCopy.userAgent))
+  let isSupported = isDesktop || (isChrome && !isMobile);
+   !isSupported
+ }
+
+
+ // This requires being loaded by defer, not as async
+open EventTarget
+let target = Webapi.Dom.document |> Webapi.Dom.Document.asEventTarget
+target->addEventListener("DOMContentLoaded", _ => {
+  if (unsupportedBrowser()) {
+    let htmlString =
+      "<div class='modal-overlay' id='unsupportedBrowser'><div class='modal'><div class='warning'><p class='title'>Dark is only fully supported on Desktop Chrome right now. We have a desktop client you can use if you prefer.</p><p class='title'>Unfortunately, as a small team, we don’t have the capacity to support other browsers at the moment. Once we’ve made the experience excellent on Chrome, we'll address cross-browser compatibility issues to support Firefox, Safari, and mobile devices.</p><p class='title'>Thanks for understanding ❤️</p></div></div></div>";
+    let div =  document->Document.createElement("div");
+    div->Element.setInnerHTML(htmlString)
+    let body =
+      document
+      ->Document.asHtmlDocument
+      ->Belt.Option.flatMap(HtmlDocument.body)
+      ->Belt.Option.getUnsafe
+    body->Element.appendChild(~child=div)
+  }
+})
+
+
+if (unsupportedBrowser()) {
+  open MutationRecord
+  // Reload page if user tries to delete overlay
+  window->Window.addEventListener("DOMContentLoaded", _ => {
+    let reload = ref(false)
+    let observer= MutationObserver.make((_records, _observer) => ())
+    let records: array<Dom.mutationRecord> = observer->MutationObserver.takeRecords
+    records|>Js.Array.forEach(record=>{
+      record->removedNodes
+      |> Webapi.Dom.NodeList.toArray
+      |> Js.Array.forEach(node=>{
+        Js.log(node)
+        // if (node["id"]== "unsupportedBrowser") {
+        //   reload := true
+        //   }
+      })
+      if (reload.contents){
+          Webapi.Dom.location->Webapi.Dom.Location.reload}
+    })
+    observer->MutationObserver.observe(
+      document->Document.documentElement,
+      {
+        "subtree": true,
+        "childList": true,
+        "attributes": true,
+        "attributeOldValue": true,
+        "characterData": true,
+      },
+    )
+    }
+  )}
+
+
+
+
+// ---------------------------
+// Analytics
+// ---------------------------
+
+
+// ---------------------------
+// Allows us capture certain keys and stop them from affecting the browser.
+// ---------------------------
+
+
+
+let stopKeys =  event =>{
+  open! KeyboardEvent
+  // Don't ever attempt to save the HTML of the page.
+ if (ctrlKey(event)|| metaKey(event)) && key(event)=="s"{
+  event -> preventDefault
+ }
+
+ // `Ctrl-K` is meant to open the omnibox on Linux, but without this preventDefault
+ // it will focus the browser's URL bar after creating the omnibox.
+
+  if ctrlKey(event)&& key(event)=="k" && Entry.getBrowserPlatform() == Linux{
```

Comment:
`window.navigator.platform `isn't found in neither webapi nor rescript's api.
`navigator.userAgentData.platform` could be an alternative but it isn't found in webapi as well.
I've used `Entry.getBrowserPlatform` but I'm not sure if it is the right choice.

Reply:
Definitely worth adding it to rescript-webapi if it's a standard feature that's missing. Then you can use the same definitions in our codebase until we can use the webapi ones.